models/spring-ai-azure-openai/src/main/java/org/springframework/ai/azure/openai/AzureOpenAiChatOptions.java

```diff
@@ -145,7 +144,6 @@ public class AzureOpenAiChatOptions implements FunctionCallingOptions, ChatOptio
 	 * default. Use the enableFunctions to set the functions from the registry to be used
 	 * by the ChatModel chat completion requests.
 	 */
-	@NestedConfigurationProperty
```

Comment:
I'm not even sure if this should work even if we kept the annotation as how does boot know what class that implements `FunctionCallback` to create?  

I tried the following in `application.yaml`

```
spring:
  ai:
    azure:
      openai:
        chat:
          options:
            function-callbacks:
              - name: "functionName1"
                description: "Description of what function1 does"
                input-type-schema: |
                  {
                    "type": "object",
                    "properties": {
                      "param1": {
                        "type": "string"
                      }
                    }
                  }
              - name: "functionName2"
                description: "Description of what function2 does"
                input-type-schema: |
                  {
                    "type": "object",
                    "properties": {
                      "param2": {
                        "type": "integer"
                      }
                    }
                  }
```

and got the following error

```
Error starting ApplicationContext. To display the condition evaluation report re-run your application with 'debug' enabled.
2024-10-30T11:41:55.878-04:00 ERROR 11409 --- [           main] o.s.b.d.LoggingFailureAnalysisReporter   : 

***************************
APPLICATION FAILED TO START
***************************

Description:

Binding to target [Bindable@7b6141f0 type = java.util.List<org.springframework.ai.model.function.FunctionCallback>, value = 'provided', annotations = array<Annotation>[@com.fasterxml.jackson.annotation.JsonIgnore(true)], bindMethod = [null]] failed:

    Property: spring.ai.azure.openai.chat.options.function-callbacks[0].description
    Value: "Description of what function1 does"
    Origin: class path resource [application.yaml] - 9:30
    Reason: The elements [spring.ai.azure.openai.chat.options.function-callbacks[0].description,spring.ai.azure.openai.chat.options.function-callbacks[0].input-type-schema,spring.ai.azure.openai.chat.options.function-callbacks[0].name,spring.ai.azure.openai.chat.options.function-callbacks[1].description,spring.ai.azure.openai.chat.options.function-callbacks[1].input-type-schema,spring.ai.azure.openai.chat.options.function-callbacks[1].name] were left unbound.
    Property: spring.ai.azure.openai.chat.options.function-callbacks[0].input-type-schema
    Value: "{
  "type": "object",
  "properties": {
    "param1": {
      "type": "string"
    }
  }
}
"
    Origin: class path resource [application.yaml] - 10:36
    Reason: The elements [spring.ai.azure.openai.chat.options.function-callbacks[0].description,spring.ai.azure.openai.chat.options.function-callbacks[0].input-type-schema,spring.ai.azure.openai.chat.options.function-callbacks[0].name,spring.ai.azure.openai.chat.options.function-callbacks[1].description,spring.ai.azure.openai.chat.options.function-callbacks[1].input-type-schema,spring.ai.azure.openai.chat.options.function-callbacks[1].name] were left unbound.
    Property: spring.ai.azure.openai.chat.options.function-callbacks[0].name
    Value: "functionName1"
    Origin: class path resource [application.yaml] - 8:23
    Reason: The elements [spring.ai.azure.openai.chat.options.function-callbacks[0].description,spring.ai.azure.openai.chat.options.function-callbacks[0].input-type-schema,spring.ai.azure.openai.chat.options.function-callbacks[0].name,spring.ai.azure.openai.chat.options.function-callbacks[1].description,spring.ai.azure.openai.chat.options.function-callbacks[1].input-type-schema,spring.ai.azure.openai.chat.options.function-callbacks[1].name] were left unbound.
    Property: spring.ai.azure.openai.chat.options.function-callbacks[1].description
    Value: "Description of what function2 does"
    Origin: class path resource [application.yaml] - 20:30
    Reason: The elements [spring.ai.azure.openai.chat.options.function-callbacks[0].description,spring.ai.azure.openai.chat.options.function-callbacks[0].input-type-schema,spring.ai.azure.openai.chat.options.function-callbacks[0].name,spring.ai.azure.openai.chat.options.function-callbacks[1].description,spring.ai.azure.openai.chat.options.function-callbacks[1].input-type-schema,spring.ai.azure.openai.chat.options.function-callbacks[1].name] were left unbound.
    Property: spring.ai.azure.openai.chat.options.function-callbacks[1].input-type-schema
    Value: "{
  "type": "object",
  "properties": {
    "param2": {
      "type": "integer"
    }
  }
}"
    Origin: class path resource [application.yaml] - 21:36
    Reason: The elements [spring.ai.azure.openai.chat.options.function-callbacks[0].description,spring.ai.azure.openai.chat.options.function-callbacks[0].input-type-schema,spring.ai.azure.openai.chat.options.function-callbacks[0].name,spring.ai.azure.openai.chat.options.function-callbacks[1].description,spring.ai.azure.openai.chat.options.function-callbacks[1].input-type-schema,spring.ai.azure.openai.chat.options.function-callbacks[1].name] were left unbound.
    Property: spring.ai.azure.openai.chat.options.function-callbacks[1].name
    Value: "functionName2"
    Origin: class path resource [application.yaml] - 19:23
    Reason: The elements [spring.ai.azure.openai.chat.options.function-callbacks[0].description,spring.ai.azure.openai.chat.options.function-callbacks[0].input-type-schema,spring.ai.azure.openai.chat.options.function-callbacks[0].name,spring.ai.azure.openai.chat.options.function-callbacks[1].description,spring.ai.azure.openai.chat.options.function-callbacks[1].input-type-schema,spring.ai.azure.openai.chat.options.function-callbacks[1].name] were left unbound.

Action:

Update your application's configuration
```

This can be researched in a separate issue as it is more important to remove the boot dependency in core than wait for solution to these sort of infrequently used options.